certbot/certbot/_internal/client.py

```diff
@@ -70,16 +69,8 @@ def acme_from_config_key(config: configuration.NamespaceConfig, key: jose.JWK,
                                     verify_ssl=(not config.no_verify_ssl),
                                     user_agent=determine_user_agent(config))
 
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore", DeprecationWarning)
-
-        client = acme_client.BackwardsCompatibleClientV2(net, key, config.server)
-        if client.acme_version == 1:
-            logger.warning(
-                "Certbot is configured to use an ACMEv1 server (%s). ACMEv1 support is deprecated"
-                " and will soon be removed. See https://community.letsencrypt.org/t/143839 for "
-                "more information.", config.server)
-        return cast(acme_client.ClientV2, client)
+    directory = messages.Directory.from_json(net.get(config.server).json())
```

Comment:
I wasn't sure whether we should create a helper in `acme` to get the `acme.messages.Directory` from a URL.

Reply:
I think my biggest question would be where to put it? `ClientNetwork` handles lower-level network operations, so that seems a bit weird. Could just be a floating utility method, but this module is way more object-oriented than the rest of the codebase, so it's a little out of place there. Could technically be a class method on `ClientV2`?

Reply:
Yes, I think a class method on `ClientV2` makes the most sense. I've added that. *Technically* we don't need the `ClientNetwork` to fetch the directory since the request is unsigned, but it does do some useful things like logging and setting the user-agent.

Reply:
Yeah, and it's kind of nice to set it up and then go do all the network stuff. I like this version!